pkg/controller/machineset/delete_policy.go

```diff
@@ -17,57 +17,102 @@ limitations under the License.
 package machineset
 
 import (
+	"math"
+	"sort"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 )
 
-type deletePriority int
+type deletePriority float64
 
 const (
-	mustDelete   deletePriority = 100
-	betterDelete deletePriority = 50
-	couldDelete  deletePriority = 20
-	// mustNotDelete deletePriority = 0
+	DeleteNodeAnnotation = "machineset.clusters.k8s.io/delete-me"
```

Comment:
(incoming bike-shed alert...)

#726 used `sigs.k8s.io/cluster-api-delete-machine`.

```
// machineDeleteAnnotationKey annotates machines to be delete among first ones
var machineDeleteAnnotationKey = "sigs.k8s.io/cluster-api-delete-machine"
```

I think there are 3 aspects here:

1) We should figure out the "correct prefix" - there are guidelines _somewhere_.  I _think_ `cluster.k8s.io` is allowed, particularly as we own that API group.
2) I don't think we need `machineset` in there - I guess we're saying this is because it is the priority for deletion when scaled down as part of a machineset?
3) `delete-me` vs `cluster-api-delete-machine` - I'd argue for the more "boring" `delete-machine`, but this is bikeshed territory.   Maybe some indication that it is a weighting, and doesn't actually cause deletion?  `deletion-priority` maybe?